rdfproxy/utils/sparql_utils.py

```diff
@@ -20,21 +20,21 @@
 """)
 
 
-def replace_query_select_clause(query: str, repl: str) -> str:
+def replace_query_select_clause(query: str, repl: str):
     """Replace the SELECT clause of a query with repl."""
-    if re.search(r"select\s.+", query, re.I) is None:
+    pattern = r"(select\s+[\s\S]*?)(?=\s+where)"
```

Comment:
why do we need the first group?
why do we need `[\s\S]*?` why not `.*?`?

Reply:
> why do we need the first group?

True, the first group isn't strictly necessary. I found the regex easier to write that way and I also find it easier to read. But I can remove the group.


>  why do we need `[\s\S]*?` why not `.*??`

`[\s\S]*?` is almost equivalent to `.*?` but is more general because it also matches linebreaks without the `re.DOTALL` flag. 

E.g. `(select\s.*?)(?=\s+where)` wouldn't match the SELECT clause in:

```
select

*
where {
?s ?p ?o . }
```

I don't know if this will even matter, because the planned query checking feature will normalize and trim incoming queries anyway (something that should definitely happen!), but even with sanitized queries I feel like the `[\s\S]` variant is more exact.

Reply:
If we can choose between `\s\S` and `.` + `re.DOTALL` I'd prefer the latter